src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -336,6 +336,32 @@ public bool IsPointer
             }
         }
 
+        public bool IsGenericParameter
+        {
+            get
+            {
+                return GetTypeFlags(TypeFlags.CategoryMask) == TypeFlags.GenericParameter;
+            }
+        }
+
+        public bool IsDefType
+        {
+            get
+            {
+                Debug.Assert(GetTypeFlags(TypeFlags.CategoryMask) <= TypeFlags.Interface == this is DefType);
+                return GetTypeFlags(TypeFlags.CategoryMask) <= TypeFlags.Interface;
+            }
+        }
+
+        public bool IsReferenceType
```

Comment:
ECMA 335 has defined meaning for "reference type" that is different from this one (it even includes unmanaged pointers). IsObjRef or something similar would be better. Also, I think it should return true for interfaces as well.